app/controllers/cart_controller.rb

```diff
@@ -11,6 +11,12 @@ def add_item
 
   def show
     @items = cart.items
+    @discounts = []
+    @items.each do |item, quantity|
```

Comment:
This should be moved as a view call which implements a model method. REFACTOR THIS. Clever logic though to find discounts.